st/clitests/awss3api_spec.py

```diff
@@ -1841,6 +1853,335 @@ def get_md5_sum(fname):
+# Expected output:
+#   foo#123, foo+123
+#   foo/ (Under COMMONPREFIXES)
+prefix = "foo"
+delimiter = "/"
+expected_versions_list = ['foo#123', 'foo+123']
+list_versions_options = {"prefix":prefix, "delimiter":delimiter,"page-size":1}
+result = AwsTest(('Aws list object versions using prefix \'%s\' and delimiter %s') % (prefix, delimiter))\
+    .list_object_versions("versionlistbucket", **list_versions_options)\
+    .execute_test()\
+    .command_is_successful()
+# Process result set
+list_versions_response = get_aws_cli_object(result.status.stdout)
+if list_versions_response is None:
+    assert False, "Failed to list object versions"
+# Get versions/common prefixes from the response
+object_versions = list_versions_response["versions"]
+common_prefixes = list_versions_response["prefix"]
+print(f"Response: {list_versions_response} Versions received: {object_versions} Common prefixes received: {common_prefixes}")
+check = all(item in object_versions for item in expected_versions_list)
+if check is False:
+    assert False, "Failed to match expected versions in the list"
+check = ("foo/" in common_prefixes)
+if check is False:
+    assert False, "Failed to match expected common prefix in the list"
+
+# Step 9: Test max-items(max-keys) and key-marker parameters
+# Step 9.1: Command:= aws s3api list-object-versions --bucket <bucket> --max-items 1 --prefix "boo" --delimiter "/"
+# Expected part1 output:
+#  boo#
+#  boo/ (Under COMMONPREFIXES)
+prefix = "boo"
+expected_versions_list_part1 = ['boo#']
+expected_versions_list_part2 = ['boo+']
+expected_common_prefixes = ['boo/']
+list_versions_options = {"prefix":prefix, "delimiter":delimiter, "max-items":1}
+result = AwsTest(('Aws list object versions using prefix \'%s\', delimiter \'%s\' and max-items %s') % (prefix, delimiter, 1))\
+    .list_object_versions("versionlistbucket", **list_versions_options)\
+    .execute_test()\
+    .command_is_successful()
+# Process result set
+list_versions_response = get_aws_cli_object(result.status.stdout)
+if list_versions_response is None:
+    assert False, "Failed to list object versions"
+# Get keys from the response
+object_versions = list_versions_response["versions"]
+common_prefixes = list_versions_response["prefix"]
+print(f"Response: {list_versions_response} Versions received: {object_versions} Common prefixes received: {common_prefixes}")
+check = all(item in object_versions for item in expected_versions_list_part1)
+if check is False:
+    assert False, "Failed to match expected versions in the list"
+check = all(item in common_prefixes for item in expected_common_prefixes)
+if check is False:
+    assert False, "Failed to match expected common prefix in the list"
+# Verify that NextToken is in the response and it is not empty
+assert (("next_token" in list_versions_response.keys()) and (len(list_versions_response["next_token"].strip()) > 0)), "NextToken is either not present or empty"
```

Comment:
Easier to read as two test assertions, and you can distinguish the failure.
```suggestion
assert "next_token" in list_versions_response, "NextToken is not present"
assert list_versions_response["next_token"].strip(), "NextToken is empty"
```

Reply:
Done